src/core/symbology-ng/qgssymbol.cpp

```diff
@@ -377,14 +375,12 @@ bool QgsSymbol::changeSymbolLayer( int index, QgsSymbolLayer *layer )
 
 void QgsSymbol::startRender( QgsRenderContext &context, const QgsFields &fields )
 {
-  delete mSymbolRenderContext;
-  mSymbolRenderContext = new QgsSymbolRenderContext( context, outputUnit(), mAlpha, false, mRenderHints, nullptr, fields, mapUnitScale() );
+  mSymbolRenderContext.reset( new QgsSymbolRenderContext( context, outputUnit(), mAlpha, false, mRenderHints, nullptr, fields, mapUnitScale() ) );
 
   QgsSymbolRenderContext symbolContext( context, outputUnit(), mAlpha, false, mRenderHints, nullptr, fields, mapUnitScale() );
 
-  QgsExpressionContextScope *scope = QgsExpressionContextUtils::updateSymbolScope( this, new QgsExpressionContextScope() );
-
-  mSymbolRenderContext->setExpressionContextScope( scope );
+  std::unique_ptr< QgsExpressionContextScope > scope( QgsExpressionContextUtils::updateSymbolScope( this, new QgsExpressionContextScope() ) );
+  mSymbolRenderContext->setExpressionContextScope( scope.release() );
```

Comment:
I've been trying to be a good c++11 citizen and always use unique_ptrs for owned pointers, even if it looks trivial. This would obviously make more sense if setExpressionContextScope used a unique_ptr argument, which hopefully it will do sometime in future....